demisto_sdk/commands/common/tests/update_id_set_test.py

```diff
@@ -2871,6 +2871,46 @@ def test_get_fields_by_script_argument(task):
 
         result = get_fields_by_script_argument(task)
         assert "field_name" in result
+    
+    
+    custom_fields_with_explicit_method ={
+        "id": "ID",
+        "scriptarguments": {
+            "customFields": {'simple': '${keys}'}
+        },
+    }
+    @staticmethod
+    def test_get_fields_by_script_argument__explicit_method(task = custom_fields_with_explicit_method):
+        """
+        Given
+            - A playbook task with custom fields that retrieves its values in an explicit way (using the ${} method instead of "get" method)
+        When
+            -  'get_fields_by_script_argument' is called
+        Then
+            -  The function should return an empty set with no errors
+        """
+        result = get_fields_by_script_argument(task)
+        assert result ==set()
+        
+
+    custom_fields_as_a_json ={
+        "id": "ID",
+        "scriptarguments": {
+            "customFields": {'complex': {'root': 'keys', 'test_key': 'tets_value','id': 'ID'}}
+        },
+    }
+    @staticmethod
+    def test_get_fields_by_script_argument__json(task = custom_fields_as_a_json):
+        """
+        Given
+            - A playbook task with custom fields value is a json
+        When
+            -  'get_fields_by_script_argument' is called
+        Then
+            -  The function should return a set with the json keys that are not in BUILT_IN_FIELDS
```

Comment:
That doesn't say much, what is `BUILT_IN_FIELDS`?
give a brief of what the expected output should be here in a "high level", rather than "code level"

Reply:
fixed